### DISCUSSION TITLE
Error in parsing GlobalVariableWriteNode when passed as command line argument.

### DISCUSSION THREAD

**Discussion by nikhilbhatt** [ruby/prism#2100] [Q&A] [answered]
Error in GlobalVariableWriteNode parse

```
bin/parse -e "$foo=1"
```

```
Errors:
[#<Prism::ParseError @message="cannot parse the expression" @location=#<Prism::Location @start_offset=0 @length=0 start_line=1>>]

AST:
@ ProgramNode (location: (1,0)-(1,2))
├── locals: []
└── statements:
    @ StatementsNode (location: (1,0)-(1,2))
    └── body: (length: 2)
        ├── @ MissingNode (location: (1,0)-(1,0))
        └── @ IntegerNode (location: (1,1)-(1,2))
            └── flags: decimal
```

The same syntax will work in file. 
```
bin/parse test.rb
```

Similar issue with GlobalVariableReadNode.

I am not sure if this is a bug or expected behaviour.

**Comment by kddnewton** [comment, marked as answer]
Your shell is probably substituting $foo before it gets to bin/parse. Try it with single quotes.

**Reply by nikhilbhatt** [reply]
So can this be updated in docs? instead of using double quotes, use single quotes for parsing?

**Reply by kddnewton** [reply]
This isn't a prism issue.


You can use double quotes just fine: bin/parse -e "@foo=1" is fine.

The issue is with string expansion in your particular shell. Some people might want to use that, I don't know it's outside the scope of prism entirely.
